### ISSUE TITLE
Server Side Batching

### ISSUE THREAD

**Issue by david-waterworth** [triton-inference-server/server#6219]
Does Triton support server-side batching?  I have an ensemble consisting of a tokeniser (Python), a classifier (torch onyx) and a post-processor that converts the classifier logits to topk labels and probabilities.

```
input [
{
    name: "text"
    data_type: TYPE_STRING
    dims: [ -1 ]
    # -1 means dynamic axis, aka this dimension may change
}
]

#output of the ensemble
output [
    {
        name: "labels"
        data_type: TYPE_INT32
        dims: [-1, 10]
    }, {
        name: "scores"
        data_type: TYPE_INT32
        dims: [-1, 10]
    }
]
```

What I'd like to do is have the tokeniser BLS chunk the input text into chunks that don't exceed the max batch size of the onyx model, then the post-processor gather the chunks and recombine them. My use case is periodic processing of large batches of data rather than lots of small requests (currently looking at hosting the models using Sagemaker Async Inference).

I think I could use this example - https://github.com/triton-inference-server/python_backend/blob/main/examples/bls_decoupled/sync_model.py  - i.e. have one BLS model that does the splitting, invokes the onyx mode with each mini-batch and then process the output chunks instead of using an ensemble.

Is this the best option or is there an alternative / easier way of doing this?

**Comment by dyastremsky** [comment]
Yes, what you are describing is supported by the [dynamic batcher.](https://github.com/triton-inference-server/server/blob/main/docs/user_guide/model_configuration.md#dynamic-batcher) Everything is done on the server side, including providing a maximum batch size as well as scattering and gathering inputs/outputs.

**Comment by dyastremsky** [comment]
Ah, Triton only batches, it does not split requests apart. There is an enhancement being investigated here: https://github.com/triton-inference-server/server/issues/4547

I'll close this issue to avoid duplication and keep all conversation around that functionality there.
